src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect } from 'react';
+import './Autocomplete.scss';
+import { Person } from '../../types/Person';
+
+interface AutocompleteProps {
+  people: Person[];
+  onSelected: (person: Person | null) => void;
+  onInputChange: () => void;
+  delay: number;
+}
+
+export const Autocomplete: React.FC<AutocompleteProps> = ({
+  people,
+  onSelected,
+  onInputChange,
+  delay,
+}) => {
+  const [inputValue, setInputValue] = useState('');
+  const [suggestions, setSuggestions] = useState<Person[]>([]);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isNoMatch, setIsNoMatch] = useState(false);
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      if (inputValue === '') {
+        setSuggestions(people);
+        setIsNoMatch(false);
+      } else {
+        const filteredPeople = people.filter(person =>
+          person.name.toLowerCase().includes(inputValue.toLowerCase()),
+        );
+
+        setSuggestions(filteredPeople);
+        setIsNoMatch(filteredPeople.length === 0 && selectedPerson === null);
+      }
+    }, delay);
+
+    return () => clearTimeout(handler);
+  }, [inputValue, people, delay, selectedPerson]);
```

Comment:
Including `selectedPerson` in the dependency array of this `useEffect` hook is unnecessary and could lead to unwanted re-renders. The effect should only depend on `inputValue`, `people`, and `delay`. Consider removing `selectedPerson` from the dependency array.